privacyidea/api/lib/prepolicy.py

```diff
@@ -1085,14 +1086,24 @@ def check_token_init(request=None, action=None):
     role = g.logged_in_user.get("role")
     admin_realm = g.logged_in_user.get("realm")
     scope = SCOPE.ADMIN
+    # We need this user, if an admin enrolls a token for a user
+    user_obj = get_user_from_param(params)
+    resolver = realm = None
     if role == ROLE.USER:
         scope = SCOPE.USER
+        realm = admin_realm
```

Comment:
Shouldn't the user have his own realm here?

Reply:
+1 -- to me it looks like we're changing behavior here, but I haven't yet fully understood this :)

Reply:
"admin_realm" contains the realm of the logged in user. (line 1087.)
I some cases the variable name is confusing.
THis is why - in case of a user - we set the user-realm name and reset the admin_realm name.